src/model/types.rs

```diff
@@ -197,3 +217,87 @@ impl From<HashMap<String, Types>> for Types {
         Self::Map(m)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::str::FromStr;
+
+    use super::*;
+
+    #[test]
+    fn parse_map_with_uuid() {
+        assert_eq!(
+            Ok((
+                "",
+                Types::Map(
+                    vec![(
```

Comment:
Idk if it's worth bringing this crate to the table, but it may help making it simpler that you're creating a `HashMap` without having to use a `Vec`: https://github.com/GrayJack/sugars